src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -0,0 +1,372 @@
+    "CellCentered": Spectral.Quadrature.CellCentered,
+    "FaceCentered": Spectral.Quadrature.FaceCentered
+}
+
+
+def basis_from_string(names):
+    return basis_dict[names] if isinstance(
+        names, str) else [basis_dict[name] for name in names]
+
+
+def quadrature_from_string(names):
+    return quadrature_dict[names] if isinstance(
+        names, str) else [quadrature_dict[name] for name in names]
+
+
+Mesh = {1: Spectral.Mesh1D, 2: Spectral.Mesh2D, 3: Spectral.Mesh3D}
+
+RegularGrid = {
+    1: Interpolation.RegularGrid1D,
+    2: Interpolation.RegularGrid2D,
+    3: Interpolation.RegularGrid3D
+}
+
+
+def interpolate_h5_file(source_file_path,
+                        mesh_args,
+                        components_to_interpolate=None,
+                        target_file_path=None,
+                        source_volume_data="/VolumeData",
+                        target_volume_data="/VolumeDataInterpolated",
+                        obs_start=-np.inf,
+                        obs_end=np.inf,
+                        obs_stride=1):
+    """
+    Interpolates an h5 file to a desired grid
+
+    The function reads data from source_volume_data inside source_file_path,
+    interpolates all components specified by components_to_interpolate to the
+    grid specified by mesh_args and writes the results into target_volume_data
+    inside target_file_path.
+    This function is designed to be used with multiprocessing.Pool which
+    requires all arguments to be serialized with pickle. This is why the mesh
+    is not passed in as an argument directly.
+
+    Parameters
+    ----------
+    source_file_path: str
+        the path to the source file where the source_volume_data is
+    mesh_args: iterable
+        the arguments used to construct a mesh to which the data is interpolated
+    components_to_interpolate: list of str, optional
+        a list of all components that are to be interpolated. accepts regular
+        expressions. By default ALL tensor components are interpolated.
+    target_file_path: str, optional
+        the path to where the interpolated data is written. By default this is
+        set to source_file_path.
+    source_volume_data: str, optional
+        the name of the .vol file inside the source file where the source data
+        can be found
+    target_volume_data: str, optional
+        the name of the .vol file inside the target file where the target data
+        is written
+    obs_start: float, optional
+        disregards all observation with observation_value before obs_start
+    obs_end: float, optional
+        disregards all observations with observation_value after obs_start
+    obs_stride: float, optional
+        will only take every obs_stride observation
+    """
+
+    if target_file_path is None:
+        target_file_path = source_file_path
+
+    source_file = spectre_h5.H5File(source_file_path, True)
+    source_vol = source_file.get_vol(source_volume_data)
```

Comment:
you essentially want to check for existence with `h5py`?

Reply:
No you can `source_file.get_groups()` to check the existence

Reply:
that makes me think, maybe we can expose these functions in a way such that we check for existence with `os` or `h5py` first before executing the spectre counterpart. ugly workaround but it doesnt seem like there will ever be another way around it. This should be possible with pybind, no?

Reply:
Yea you're right it would be good to do this on the side of the bindings, basically working around the broken error propagation (or fix that). I think we don't have to call into Python for that, but we can do the check in C++ in the bindings. I.e. use `Utilities/FileSystem.hpp:check_if_file_exists` and `File::get_groups`.

Reply:
yeah thats probably better. I can try and implement that. Fixing the error propagation is probably a huge if not impossible task due to parallel design. As you said, since most funcs are marked `noexcept` the error wont get very far atm.

Reply:
👍 sounds good (put it in a separate PR please and link the issue https://github.com/sxs-collaboration/spectre/issues/2312 in a code comment)